nav2_collision_monitor/src/kinematics.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2022 Samsung Research Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_collision_monitor/kinematics.hpp"
+
+#include <cmath>
+
+namespace nav2_collision_monitor
+{
+
+void transformPoint(const Velocity & velocity, const double dt, Point & point)
```

Comment:
I don't think any of these functions are named properly, this function does not transform a point, it is manipulating a pre-existing point by the velocity and time, not simply projecting it. Even if just projecting, `transform` is not what is happening here. As well for the other implementation of `transformPoint` below which I'm not really sure what it is doing

Reply:
This implementation to be removed in further update. Another implementation:
```
void transformPoints(const Pose & pose, std::vector<Point> & points);
```
do a transformation of points coordinates from the frame coinciding with the `(0,0)` origin -> to the frame, which origin is equal to `pose`. This is non-existing in ROS frame used in simulation, new per each step. Therefore, I do not want to create a new TF frame on each step, as it will lead to unnecessary performance overheads.

Inherently, this routine do a transformation of non-moving points from one frame to another, so I do not know what name to think of better than `transformPoints()` ?

Reply:
[Done](https://github.com/ros-planning/navigation2/pull/2982/files#diff-a9b2ec393371be6ada3bd9d02053c84a46307b5ab39290cf2b01cc2951576313R23-R29)